sstables/mx/reader.cc

```diff
@@ -674,7 +677,7 @@ class data_consume_rows_context_m : public data_consumer::continuous_data_consum
     // becomes false when we yield in the main coroutine, although we don't need to consume
     // more data buffers to continue, switch back to true afterwards
     bool _consuming = true;
-    mp_row_consumer_m& _consumer;
+    Consumer& _consumer;
```

Comment:
In fact I had vague plans to reduce the levels of hierarchy in this area. I guess the hierarchy is merited here since the validator will want more information exported than the regular readers.